From 9bcf6ab4f14ef51f88eb518837b407d82d0b0ebf Mon Sep 17 00:00:00 2001
From: "Tan, Tee Min" <tee.min.tan@intel.com>
Date: Wed, 24 Apr 2019 22:04:32 +0800
Subject: [PATCH 49/52] REVERTME: Workaround for TSN AIC driver unload issue

Temporary disables CONFIG_HWMON in marvell10g.c

While during driver unload, kernel will call to mv3310_hwmon_disable()
to disable the thermal monitoring, but at that moment, the phydev
already been deleted at the earlier stage.

Then, it will causes the unsuccessful unload and showing dump stack
because the kernel was not able to handle NULL pointer.

Signed-off-by: Tan, Tee Min <tee.min.tan@intel.com>
---
 drivers/net/phy/marvell10g.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/marvell10g.c b/drivers/net/phy/marvell10g.c
index 1813aae..8b4af85 100644
--- a/drivers/net/phy/marvell10g.c
+++ b/drivers/net/phy/marvell10g.c
@@ -71,7 +71,7 @@ struct mv3310_priv {
 	char *hwmon_name;
 };
 
-#ifdef CONFIG_HWMON
+#if 0
 static umode_t mv3310_hwmon_is_visible(const void *data,
 				       enum hwmon_sensor_types type,
 				       u32 attr, int channel)
-- 
2.7.4

